third_party/model_api/model_api.bzl

```diff
@@ -14,13 +14,57 @@
 # limitations under the License.
 #
 
-def _impl(repository_ctx):
-    http_proxy = repository_ctx.os.environ.get("http_proxy", "")
-    https_proxy = repository_ctx.os.environ.get("https_proxy", "")
-    # Note we need to escape '{/}' by doubling them due to call to format
+def _is_windows(ctx):
+    return ctx.os.name.lower().find("windows") != -1
+
+def _get_windows_build_file():
+    build_file_content = """
+load("@rules_foreign_cc//foreign_cc:cmake.bzl", "cmake")
+
+visibility = ["//visibility:public"]
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["model_api/cpp/**"]),
+    visibility = ["//visibility:public"],
+)
+
+cmake(
+    name = "model_api_cmake",
+    build_args = [
+        "--verbose",
+        "-j 24",
+    ],
+    cache_entries = {{
+        "CMAKE_POSITION_INDEPENDENT_CODE": "ON",
+        "OpenVINO_DIR": "C:/opt/intel/openvino_2024/runtime/cmake",
+        "OpenCV_DIR": "C:/opt/opencv/build",
```

Comment:
Are this paths correct for Windows? In `WORKSPACE` you use double backslash `\\` and here the separator is `/`.

Reply:
Yes, in bzl files it takes care of backslash / slash conversions.